src/plone/restapi/services/discussion/conversation.py

```diff
@@ -0,0 +1,203 @@
+            raise Unauthorized()
+
+        # Fake request data
+        body = json_body(self.request)
+        for key, value in body.items():
+            self.request.form['form.widgets.' + key] = value
+
+        form = EditCommentForm(comment, self.request)
+        form.__parent__ = form.context.__parent__.__parent__
+        form.update()
+
+        action = form.actions['comment']
+        data, errors = form.extractData()
+        if errors:
+            raise BadRequest({'errors': [err.error for err in errors]})
+
+        comment.modification_date = datetime.utcnow()
+        form.handleComment(form=form, action=action)
+
+        self.request.response.setStatus(204)
+        if 'location' in self.request.response.headers:
+            del self.request.response.headers['location']
+
+    def edit_comment_allowed(self):
+        # Check if editing comments is allowed in the registry
+        registry = queryUtility(IRegistry)
+        settings = registry.forInterface(IDiscussionSettings, check=False)
+        return settings.edit_comment_enabled
+
+    def can_edit(self, reply):
+        """Returns true if current user has the 'Edit comments'
+        permission.
+        """
+        return getSecurityManager().checkPermission(
+            'Edit comments', aq_inner(reply)
+        )
+
+
+@implementer(IPublishTraverse)
+class CommentsDelete(Service):
+    comment_id = None
+
+    def publishTraverse(self, request, name):
+        self.comment_id = long(name)
+        return self
+
+    def reply(self):
+        conversation = IConversation(self.context)
+        if not self.comment_id:
+            raise BadRequest("Comment id is a required part of the url")
+
+        if self.comment_id not in conversation:
+            self.request.response.setStatus(404)
+            return
+
+        comment = conversation[self.comment_id]
+
+        # Permission checks
+        can_delete = self.can_delete(comment)
+        doc_allowed = self.delete_own_comment_allowed()
+        delete_own = doc_allowed and self.could_delete_own(comment)
+        if not (can_delete or delete_own):
+            raise Unauthorized()
+
+        del conversation[self.comment_id]
+        self.request.response.setStatus(204)
+
+    # Helper functions copied from p.a.discussion's viewlet to support Plone 4
+    def can_review(self):
+        """Returns true if current user has the 'Review comments' permission.
+        """
+        return getSecurityManager().checkPermission('Review comments',
+                                                    aq_inner(self.context))
+
+    def can_delete(self, reply):
+        """Returns true if current user has the 'Delete comments'
+        permission.
+        """
+        if PLONE4:
+            return self.can_review()
+        return getSecurityManager().checkPermission(
+            'Delete comments', aq_inner(reply)
+        )
+
+    def delete_own_comment_allowed(self):
+        if PLONE4:
```

Comment:
@jaroel could you elaborate this please? And then add a comment to the code so others with the same question understand why Plone 4 and 5 are different here?